.github/workflows/test.yml

```diff
@@ -39,6 +39,10 @@ jobs:
       with:
         ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.event.ref }}
 
+    - uses: actions/setup-python@v5
```

Comment:
this is a no-op (I know that's not inuitive) + is easy to break in the future; would remove

Reply:
What about about for catching pip packages?

Reply:
the `cache: pip` stanza keys on a `requirements.txt` file and crashes if it can't find one

the reason it's not crashing in our case is that we're not passing `with: python-version`, so the action is defaulting to the system python + short-circuiting

Reply:
Cool, removed setup python and apt updates